lib/mdm/host/operating_system_normalization.rb

```diff
@@ -232,7 +231,7 @@ def recog_matches_for_service(s)
       end
       res = Recog::Nizer.match(rdb, banner)
       matches << res if res
-    end
+    end if service_match_keys.has_key?(s.name)
```

Comment:
I'd prefer not having trailing modifiers at then end of a `do end` block as it is easy to miss.